File: ReactNativeApp/ios/Modules/SSVC/Games/Juggling/JugglingBuilder.swift
//
//  JugglingBuilder.swift
//  jogo
//
//  Created by arham on 18/11/2021.
//

import Foundation

class JugglingBuilder: SSVCBuilder {
  
  internal static func buildState() -> JugglingState {
    return JugglingState()
  }
  
  internal static func build(ballController: BallController?,
                             personController: PersonController?,
                             pointController: PointsController?) -> SSVC<JugglingState, JugglingView, JugglingController> {
    let state = self.buildState()
    let view = JugglingView(state: state)
    let controller = JugglingController(state: state,
                                        ballController: ballController,
                                        personController: personController,
                                        pointController: pointController)
    return SSVC(state: state, stateView: view, controller: controller)
  }
  
}

File: ReactNativeApp/android/app/src/main/java/com/components/games/instances/juggling/JugglingBuilder.java
package com.components.games.instances.juggling;

import com.components.basic.points.PointsController;
import com.components.basic.streak.StreakController;
import com.components.games.base.GameContext;
import com.ssvc.SSVC;

public class JugglingBuilder {

    static private JugglingState buildState() {
        return new JugglingState();
    }

    public static SSVC<JugglingState, JugglingView, JugglingController> buildJugglingGame(PointsController pointsController, StreakController streakController) {
        JugglingState state = buildState();
        JugglingView jugglingView = new JugglingView(state, GameContext.getContext().ball.state);
        JugglingController jugglingController = new JugglingController(state, GameContext.getContext().ball.controller,
                GameContext.getContext().person.controller, pointsController,streakController);

        return new SSVC<>(state, jugglingView, jugglingController);
    }

}
